### PR TITLE
Add tests for identify

### PR DESCRIPTION
**Pull request by ralexstokes**
Basic tests for identify.

PR on top of #337 and #345.

I'll rebase once these two are merged -- only the last two commits are part of this PR

### CODE REVIEW

File: tests/identity/identify/test_protocol.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+@pytest.mark.asyncio
+async def test_identify_protocol():
+    transport_opt_list = [["/ip4/127.0.0.1/tcp/0"], ["/ip4/127.0.0.1/tcp/0"]]
+    (host_a, host_b) = await set_up_nodes_by_transport_opt(transport_opt_list)
+
+    addr = host_a.get_addrs()[0]
+    info = info_from_p2p_addr(addr)
+    await host_b.connect(info)
```
Comment (mhchia) [inline tests/identity/identify/test_protocol.py]: I think we can use `host_pair_factory` in `tests/factories.py` here, which makes two connected `BasicHost`s.

File: tests/host/test_ping.py [hunk 2]
```diff
@@ -3,28 +3,16 @@
...
-    # as a forced reminder to fix this test by removing the calls to
-    # `_add_ping_to`
-    assert host_a.get_mux().handlers == {}
-    assert host_b.get_mux().handlers == {}
-    _add_ping_to(host_a)
-    _add_ping_to(host_b)
-
     addr = host_a.get_addrs()[0]
     info = info_from_p2p_addr(addr)
     await host_b.connect(info)
```
Comment (mhchia) [inline tests/host/test_ping.py]: Ditto, we can use `host_pair_factory` from `tests/factories.py` as well.

File: tests/identity/identify/test_protocol.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    stream = await host_b.new_stream(host_a.get_id(), (ID,))
+    response = await stream.read()
+    await stream.close()
+
+    identify_response = Identify()
+    identify_response.ParseFromString(response)
+    assert identify_response == _mk_identify_protobuf(host_a)
```
Comment (mhchia) [inline tests/identity/identify/test_protocol.py]: I just came up with that we should close the hosts at the end of the test.

**Comment by ralexstokes** [comment]
@mhchia thanks for the feedback, i'll merge shortly

**Pull request merged**